tests/unit/hive_metastore/test_locations.py

```diff
@@ -718,3 +719,184 @@ def my_side_effect(path, **_):
+            return [z_dir_json]
+        return None
+
+    client.dbutils.fs.ls.side_effect = my_side_effect
+    backend = MockBackend(
+        rows={
+            '`hive_metastore`.`test`.`tables`': [],
+            '`test`.`mounts`': MOUNT_STORAGE[("/mnt/test_mount", "adls://bucket/")],
+        }
+    )
+    mounts = Mounts(backend, client, "test")
+    results = TablesInMounts(backend, client, "test", mounts).snapshot()
+    assert results == [
+        Table(
+            "hive_metastore",
+            "mounted_test_mount",
+            "table1",
+            "EXTERNAL",
+            "JSON",
+            "adls://bucket/entity/domain/table1",
+        ),
+        Table(
+            "hive_metastore",
+            "mounted_test_mount",
+            "domain",
+            "EXTERNAL",
+            "CSV",
+            "adls://bucket/entity/domain",
+        ),
+        Table(
+            "hive_metastore",
+            "mounted_test_mount",
+            "z_dir",
+            "EXTERNAL",
+            "JSON",
+            "adls://bucket/entity/domain/z_dir",
+        ),
+    ]
+
+
+def test_mount_dont_list_partitions():
+    client = create_autospec(WorkspaceClient)
+
+    first_folder = FileInfo("dbfs:/mnt/test_mount/entity/", "entity/", 0, "")
+    first_first_partition = FileInfo("dbfs:/mnt/test_mount/entity/xxx=yyy/", "xxx=yyy/", 0, "")
+    first_first_partition_files = FileInfo("dbfs:/mnt/test_mount/entity/xxx=yyy/1.parquet", "1.parquet", "", "")
+    first_second_partition = FileInfo("dbfs:/mnt/test_mount/entity/xxx=zzz/", "xxx=zzz/", 0, "")
+    first_second_partition_files = FileInfo("dbfs:/mnt/test_mount/entity/xxx=zzz/1.parquet", "1.parquet", "", "")
+    misplaced_json = FileInfo("dbfs:/mnt/test_mount/entity/xxx=zzz/misplaced.json", "misplaced.json", "", "")
+    first_delta_log = FileInfo("dbfs:/mnt/test_mount/entity/_delta_log/", "_delta_log/", 0, "")
+
+    def my_side_effect(path, **_):
+        if path == "/mnt/test_mount":
+            return [first_folder]
+        if path == "dbfs:/mnt/test_mount/entity/":
+            return [first_delta_log, first_first_partition, first_second_partition]
+        if path == "dbfs:/mnt/test_mount/entity/xxx=yyy/":
+            return [first_first_partition_files, misplaced_json]
+        if path == "dbfs:/mnt/test_mount/entity/xxx=zzz/":
+            return [first_second_partition_files]
+        return None
+
+    client.dbutils.fs.ls.side_effect = my_side_effect
+    backend = MockBackend(
+        rows={
+            '`hive_metastore`.`test`.`tables`': [],
+            '`test`.`mounts`': MOUNT_STORAGE[("/mnt/test_mount", "adls://bucket/")],
+        }
+    )
+    mounts = Mounts(backend, client, "test")
+    results = TablesInMounts(backend, client, "test", mounts).snapshot()
+    assert len(results) == 1
+    assert results[0].table_format == "DELTA"
+    assert results[0].is_partitioned
+
+
+def test_mount_infinite_loop():
```

Comment:
What should happen with an inifinte loop?

Reply:
It should be caught before going to PROD! This was a bug fix. I created this test to help identify an fix an infinite loop that sometimes appeared for files that had an `=` in them.